src/file.cpp

```diff
@@ -336,14 +337,30 @@ namespace libtorrent
 		return int(size);
 	}
 
-#ifdef TORRENT_WINDOWS
-	std::string convert_separators(std::string p)
+	template <typename T>
+	std::string convert_from_native_path(T* s);
+
+	template <>
+	std::string convert_from_native_path(char const* s) { return convert_from_native(s); }
```

Comment:
It doesn't seem right to me that these are template specializations. Is there a reason for not just making them overloads? That typically makes for much more intuitive behavior